backend/prompt_studio/prompt_studio_index_manager/models.py

```diff
@@ -78,3 +88,28 @@ class Meta:
                 name="unique_document_manager_profile_manager",
             ),
         ]
+
+
+# Function will be executed every time an instance of IndexManager is deleted.
+@receiver(post_delete, sender=IndexManager)
+def perform_vector_db_cleanup(sender, instance, **kwargs):
+    """Signal to perform vector db cleanup."""
+    logger.info("Performing vector db cleanup")
+    try:
+        # Get the index_ids_history to clean up from the vector db
+        index_ids_history = json.loads(instance.index_ids_history)
+        vector_db_instance_id = str(instance.profile_manager.vector_store.id)
+        # Generate a run_id
+        org_schema = connection.tenant.schema_name
+        util = PromptIdeBaseTool(log_level=LogLevel.INFO, org_id=org_schema)
+        vector_db = VectorDB(
+            tool=util,
+            adapter_instance_id=vector_db_instance_id,
+        )
+        for index_id in index_ids_history:
+            logger.info(f"Deleting from VectorDB - index id: {index_id}")
```

Comment:
NIT: This might be noisy, consider making it debug. You might have to fix the line length for this log if you take up the suggestion
```suggestion
            logger.debug(f"Deleting index id: {index_id} from vectorDB {vector_db_instance_id}")
```